tests/test_all_integration.py

```diff
@@ -359,7 +357,7 @@ def test_communication_manager_integration(new_session_event, new_session_key, c
     assert communication_manager.optimizer.curr_averages_this_round == 0, \
         "Either did not hear anything, or heard too much!"
     # now we should be ready to train
-    assert communication_manager.optimizer.job.job_type == JobTypes.JOB_TRAIN.name, \
-        "Should be ready to train!"
+    # assert communication_manager.optimizer.job.job_type == JobTypes.JOB_TRAIN.name, \
+    #     "Should be ready to train!"
```

Comment:
Why are all these type checks commented?

Reply:
These type checks worked because the optimizer had a job instance attribute which essentially "forecasted" what the optimizer was going to do. However, with this refactor the optimizer is more or less stateless so it doesn't maintain "intent" (optimizer.job.job_type)

Reply:
Sorry sorry I meant commented out vs deleted. Like why not delete them?

Reply:
It's still possible to check these, technically. It's just very slow -so these are left in so that someone can know what they are supposed to be looking for if they want to check.

Reply:
Okay sounds good